packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -306,12 +307,12 @@ class AOTSnapshotter {
     final CompilerOutput compilerOutput = await kernelCompiler.compile(
       sdkRoot: artifacts.getArtifactPath(Artifact.flutterPatchedSdkPath),
       mainPath: mainPath,
-      outputFilePath: fs.path.join(outputPath, 'app.dill'),
+      outputFilePathNoExtension: fs.path.join(outputPath, 'app'),
```

Comment:
Is the goal of this to keep filename neat? I would just keep adding '.track.dill' suffix if needed.